### ISSUE TITLE
Multiple IPs

### ISSUE THREAD

**Issue by aaronryank** [aaronryank/Forked#4]
I plan on making Forked have multiple IPs. Is this a good idea?

**Comment by aaronryank** [comment]
Multiple-IP commands:

 - Commands to create IP at current point, beginning
 - Commands to jump to previous IP, next IP, 0-indexed IP, Top of stack representing an IP index

**Comment by catb0t** [comment]
This sounds cool. What when they cross paths? They just continue? Bounce off one another, or jump to a certain location given the stack value? Perhaps [set your printer on fire](https://en.wikipedia.org/wiki/Lp0_on_fire)?

**Comment by aaronryank** [comment]
@catb0t I was planning on having the multiple IPs run asynchronously. i.e. only one going at a time.

However, this is why I put the spec on GitHub before implementing it - so I can get this kind of feedback. Do you think the IPs should be running synchronously, or should I wait to implement that in Fred (Forked/Thread, where `:` performs the system call `fork()`)?

**Comment by catb0t** [comment]
Yeah, maybe synchronous IPs make the most sense in Fred. By which mechanism will a given IP be switched to?

**Comment by aaronryank** [comment]
@catb0t define "mechanism". Do you mean command? Or the internal functions?

**Comment by aaronryank** [comment]
@catb0t define "mechanism". Do you mean the command? Or internal code/functions?
